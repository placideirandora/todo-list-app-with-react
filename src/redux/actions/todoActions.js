// Action types
export const ADD_TODO = 'ADD_TODO';
export const REMOVE_TODO = 'REMOVE_TODO';

// Action creators
export const addTodo = content => {
  return {
    type: ADD_TODO,
    payload: content
  };
};

export const removeTodo = id => {
  return {
    type: REMOVE_TODO,
    payload: id
  };
};
