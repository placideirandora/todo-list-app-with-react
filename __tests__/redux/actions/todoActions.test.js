import configureStore from 'redux-mock-store';
import * as todoActions from '../../../src/redux/actions/todoActions';

const mockStore = configureStore();
const store = mockStore();

describe('Todo Action Test Cases', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('should dispatch the correct action and payload for adding a todo', () => {
    store.dispatch(todoActions.addTodo('Read Books'));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should dispatch the correct action and payload for completing a todo', () => {
    store.dispatch(todoActions.removeTodo(1));
    expect(store.getActions()).toMatchSnapshot();
  });
});
