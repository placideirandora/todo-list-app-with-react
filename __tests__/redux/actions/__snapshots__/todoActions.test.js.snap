// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Action Test Cases should dispatch the correct action and payload for adding a todo 1`] = `
Array [
  Object {
    "payload": "Read Books",
    "type": "ADD_TODO",
  },
]
`;

exports[`Todo Action Test Cases should dispatch the correct action and payload for completing a todo 1`] = `
Array [
  Object {
    "payload": 1,
    "type": "REMOVE_TODO",
  },
]
`;
